### ISSUE TITLE
onNodesDelete/onEdgesDelete triggered everytime deleteKeyCode is pressed

### ISSUE THREAD

**Issue by alessandropisu** [wbkd/react-flow#1987]
### Describe the Bug

The callbacks **onNodesDelete** and **onEdgesDelete** in _Uncontrolled mode_ are triggered not only when some nodes/edges are selected and user press the deleteKeyCode (Backspace for example) but also when there aren't elements selected so basically everytime deleteKeyCode is pressed with focus on <ReactFlow /> component

### Steps to Reproduce the Bug or Issue

Press on <ReactFlow /> to have focus on it, then press the key set to delete elements (default is Backspace). You will receive as callback value for onNodesDelete/onEdgesDelete an empty array

### Expected behavior

No callback events should be triggered if no elements are deleted

### Platform

- OS: macOS
- Browser: Chrome
- Version: v10

**Comment by AndrewRayCode** [comment]
On version 10, pressing the backspace key also fires `onEdgesChange` multiple times and if you hold down backspace, triggers 
```
Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.
    at StoreUpdater (webpack-internal:///./node_modules/react-flow-renderer/dist/esm/index.js:1027:20)
```

**Comment by moklick** [comment]
thanks for the report @alessandropisu ! This is fixed in v10.0.5
